Neos.Neos/Classes/Domain/Service/WorkspaceService.php

```diff
@@ -122,44 +132,64 @@ public function createRootWorkspace(ContentRepositoryId $contentRepositoryId, Wo
                 ContentStreamId::create()
             )
         );
-        $this->metadataAndRoleRepository->addWorkspaceMetadata($contentRepositoryId, $workspaceName, $title, $description, WorkspaceClassification::ROOT, null);
-    }
-
-    /**
-     * Create the "live" root workspace with the default role assignment (users with the role "Neos.Neos:LivePublisher" are collaborators)
-     */
-    public function createLiveWorkspaceIfMissing(ContentRepositoryId $contentRepositoryId): void
-    {
-        $contentRepository = $this->contentRepositoryRegistry->get($contentRepositoryId);
-        $workspaceName = WorkspaceName::forLive();
-        $liveWorkspace = $contentRepository->findWorkspaceByName($workspaceName);
-        if ($liveWorkspace !== null) {
-            // live workspace already exists
-            return;
+        $this->metadataAndRoleRepository->addWorkspaceMetadata($contentRepositoryId, $workspaceName, $title, $description, WorkspaceClassification::ROOT, ownerUserId: null);
+        foreach ($assignments as $assignment) {
+            $this->metadataAndRoleRepository->assignWorkspaceRole($contentRepositoryId, $workspaceName, $assignment);
```

Comment:
A single, atomic, operation for the assignments would be nice, should we add some `assignWorkspaceRoles()` method to the repository or even allow for `WorkspaceRoleAssignments|WorkspaceRoleAssignment`?

Reply:
okay looked a bit into it, multi insert is not abstracted by dbal and i dont want to write native sql here and risk losing support for postgres or something unexpected ^^: https://github.com/doctrine/dbal/pull/682

multiple operations dont seem as clean but to make it truly atomic the metadata should also be added in the same commit. Maybe we could use transactions?

The operation will never be fully atomic as the content repository workspace is already created - but i guess we could optimise the metadata side.

Reply:
implemented via transactional closure via [TASK: Make workspace role and metadata creation atomic](https://github.com/neos/neos-development-collection/pull/5334/commits/08d01e133bc554b1ac9139176a2d6f20f9f0fb54)